src/arrays/qucoeffs.jl

```diff
@@ -0,0 +1,69 @@
+import Base: transpose,
+    ctranspose,
+    conj,
+    size,
+    ndims,
+    length,
+    getindex,
+    setindex!
+
+############
+# QuCoeffs #                 
+############
+    abstract ConjBool{Conj}
+    abstract TranBool{Tran}
+
+    type QuCoeffs{Conj,Tran,T,N,A}
+        arr::A  
+        conj::Type{ConjBool{Conj}}
+        tran::Type{TranBool{Tran}}
+    end
+
+    function QuCoeffs{T,N,Conj,Tran}(arr::AbstractArray{T,N}, 
+                                     conj::Type{ConjBool{Conj}}, 
+                                     tran::Type{TranBool{Tran}})
+        return QuCoeffs{Conj,Tran,T,N,typeof(arr)}(arr, conj, tran)
+    end
+
+    QuCoeffs(arr) = QuCoeffs(arr, ConjBool{false}, TranBool{false})
+    
+    typealias CoeffsVector{Conj,Tran,T,A} QuCoeffs{Conj,Tran,T,1,A}
+    typealias CoeffsMatrix{Conj,Tran,T,A} QuCoeffs{Conj,Tran,T,2,A}
+
+    typealias ConjVector{Tran,T,A} CoeffsVector{true,Tran,T,A}
+    typealias ConjMatrix{Tran,T,A} CoeffsMatrix{true,Tran,T,A}
+
+    typealias TranVector{Conj,T,A} CoeffsVector{Conj,true,T,A}
+    typealias TranMatrix{Conj,T,A} CoeffsMatrix{Conj,true,T,A}
+
+    typealias AdjVector{T,A} CoeffsVector{true,true,T,A}
+    typealias AdjMatrix{T,A} CoeffsMatrix{true,true,T,A}
+
+    ########################
+    # Array-like Functions #
+    ########################
+    size(qc::QuCoeffs) = size(qc.arr)
+    size(qc::QuCoeffs, i...) = size(qc.arr, i...)
+    size(tm::TranMatrix) = reverse(size(tm.arr))
+    size(tm::TranMatrix, i) = size(tm)[i]
+    ndims(qc::QuCoeffs) = length(size(qc))
+    length(qc::QuCoeffs) = prod(size(qc))
+
+    apply_conj(i::Complex, ::Type{ConjBool{true}}) = conj(i)
+    apply_conj(i::Complex, qc::QuCoeffs) = apply_conj(i, qc.conj)
+    apply_conj(i, conj) = i
+
+    getindex(cv::CoeffsVector, i) = apply_conj(cv.arr[i], cv)
+    getindex(cv::CoeffsMatrix, i, j) = apply_conj(cv.arr[i,j], cv)
+    getindex(cv::TranMatrix, i, j) = apply_conj(cv.arr[j,i], cv)
+
+    setindex!(cv::CoeffsVector, x, y) = setindex!(cv, apply_conj(x, cv), y)
+    setindex!(cv::CoeffsMatrix, x, y, z) = setindex!(cv, apply_conj(x, cv), y, z)
+    setindex!(cv::TranMatrix,  x, y, z) = setindex!(cv, apply_conj(x, cv), z, y)
```

Comment:
One of the advantages of not subtyping QuCoeffs to AbstractArray is that we can more easily enforce our own indexing scheme (and also are freed from having to make a bunch of functions to resolve ambiguities). I didn't implement indexing for N>2. @Jutho: You said TensorToolbox covers a more general dual indexing scheme for arbitrary N, yeah? Do you think it would make sense to focus our efforts with QuCoeffs on N<=2 case until TensorToolbox reaches a suitable state? I wouldn't want to repeat work :)
  
I would also be down to contribute to TensorToolbox if there was a little bit of documentation to help new contributors get going; no pressure, though. I, myself, won't have time to focus on anything besides QuBase/QuDirac until March (after SQuINT) anyway.